### DISCUSSION TITLE
AppSet selector based on cluster Kubernetes version

### DISCUSSION THREAD

**Discussion by tamcore** [argoproj/argo-cd#13059] [Q&A]
Is it possible to include a cluster's Kubernetes version as a condition in an AppSet?

For example,
```
argocd cluster get mycluster
```
already includes a line like
```
serverVersion: "1.26"
```
So we were wondering, if we could make use of that in an ApplicationSet? 

Unfortunately we have 2 AppSets that are, as of now, manually maintained based on the List Generator. Having the option to automatically select clusters by their K8s version, would be a big help here.

**Comment by jgwest** [comment]
Hi @tamcore, not AFAIK. I don't believe that value is surfaced as a parameter by any of the supported generators of ApplicationSet.
